manager/integration/tests/test_snapshot_check.py

```diff
@@ -0,0 +1,425 @@
+
+
+def get_checksum_in_xattr(disk_path):
+    try:
+        value = os.getxattr(disk_path, "user.longhorn.metadata",
+                            follow_symlinks=False)
+    except OSError as e:
+        print(e)
+        raise
+
+    metadata = json.loads(str(value, 'utf-8'))
+    return metadata["checksum"]
+
+
+def check_snapshot_checksums_not_set(volume):
+    value = volume.snapshotList()
+    snapshots = value.data
+    for s in snapshots:
+        if s.checksum != "":
+            return False
+
+    return True
+
+
+def get_local_host_replica_data_path(volume):
+    lht_hostId = get_self_host_id()
+
+    dataPath = ""
+    replicas = volume.replicas
+    for r in replicas:
+        if r.hostId == lht_hostId:
+            dataPath = r.dataPath
+            break
+
+    return dataPath
+
+
+def get_checksum_from_snapshot_disk_file(data_path, snapshot_name):
+    exec_cmd = ["java", "-jar", "/jacksum.jar", "-a", "crc64_go-iso",
+                os.path.join(data_path,
+                             "volume-snap-" + snapshot_name + ".img")]
+    try:
+        output = subprocess.check_output(exec_cmd)
+    except subprocess.CalledProcessError as e:
+        print(e.output)
+        return ""
+
+    data = str(output, 'utf-8')
+    fields = data.split(' ')
+    return fields[0]
+
+
+def create_snapshots(client, volume, data_size):  # NOQA
+    dev = get_volume_endpoint(volume)
+
+    assert write_device_random_data(dev, data_size)
+    snap1 = create_snapshot(client, volume.name)
+
+    assert write_device_random_data(dev, data_size)
+    snap2 = create_snapshot(client, volume.name)
+
+    assert write_device_random_data(dev, data_size)
+    snap3 = create_snapshot(client, volume.name)
+
+    snapshots = volume.snapshotList()
+    snapMap = {}
+    for snap in snapshots:
+        snapMap[snap.name] = snap
+
+    assert snapMap[snap1.name].name == snap1.name
+    assert snapMap[snap1.name].removed is False
+    assert snapMap[snap2.name].name == snap2.name
+    assert snapMap[snap2.name].parent == snap1.name
+    assert snapMap[snap2.name].removed is False
+    assert snapMap[snap3.name].name == snap3.name
+    assert snapMap[snap3.name].parent == snap2.name
+    assert snapMap[snap3.name].removed is False
+
+
+def write_device_random_data(dev, size):
+    exec_cmd = ["dd", "if=/dev/urandom", "of=" + dev,
+                "bs=1M", "count=" + str(size)]
+    try:
+        subprocess.check_output(exec_cmd)
+    except subprocess.CalledProcessError as e:
+        print(e.output)
+        return False
+    return True
+
+
+def check_hash_calculated_and_with_immediate_hash(client, volume_name, snapshot_data_integrity):  # NOQA
+    """
+    1. Create and attach a volume
+    2. Create snapshots
+    3. Check snapshots' checksums are calculated and set and also
+       check the checksums of the snapshot disk files are matched
+       with the values from the api
+    4. Check snapshots' checksums are calculated by periodic snapshot
```

Comment:
This step is not implemented in the test. Should we first clean up the volume's `checksum` manually and see if the periodic snapshot verification set the `checksum` again?

Reply:
Ah, my bad. The 4. is actually done in the step 3. 

> hould we first clean up the volume's checksum manually and see if the periodic snapshot verification set the checksum again?

If the checksum is somehow cleaned up, the checksum will be re-calculated by the periodic snapshot verification.

Reply:
Updated the steps. Thank you.